corepkgs/buildenv.pl

```diff
@@ -110,12 +110,13 @@ sub addPkg {
     my $propagatedFN = "$pkgDir/nix-support/propagated-user-env-packages";
     if (-e $propagatedFN) {
         open PROP, "<$propagatedFN" or die;
-        my $propagated = <PROP>;
-        close PROP;
-        my @propagated = split ' ', $propagated;
-        foreach my $p (@propagated) {
-            $postponed{$p} = 1 unless defined $done{$p};
+        while (my $propagated = <DATA>) {
```

Comment:
I think you meant `<PROP>` here instead of `<DATA>`.

Slightly simpler is to read the whole file and split it on arbitrary whitespace:
```
local $/;
foreach my $p (split " ", <PROP>) {
  $postponed{$p} = 1 unless defined $done{$p};
}
```